debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+                    builder.field(structField.name(), TRANSLATORS.get(TypeConverters.get(structFieldSchemaName)).typeSchema());
+                }
+            } else {
+                if (nestedStructField == null || nestedStructField.equals("")) {
+                    builder.field(structField.name(), OPTIONAL_TRANSLATORS.get(structField.schema().type().toString()).optionalSchemaNonTimestamp());
+                } else {
+                    builder.field(structField.name(), structField.schema());
+                }
+            }
+        }
+        return builder;
+    }
+
+    /**
+     * Creates the new field value of the nested struct that matches with config.field_type.
+     * Avoids all the null value fields and converts the desired field value that matches to the config parameter field_type.
+     *
+     * @param field               the input nested Struct field.
+     * @param value               the input nested Struct field value.
+     * @param nestedUpdatedSchema updated field schema with target formats.
+     */
+    private Struct NestedStructFieldValueUpdate(Field field, Struct value, Field nestedUpdatedSchema) {
+        Struct updatedNestedStructValue = new Struct(nestedUpdatedSchema.schema());
+        for (Field nestedField : ((Struct) value.get(field)).schema().fields()) { //try to convert ((Struct) value.get(field)).get(nestedField) into a single field rather than computing every single time
+            final Object updatedNestedFieldValue;
+            final Object nestedFieldValue = ((Struct) value.get(field)).get(nestedField);
+            if (nestedFieldValue == null || nestedFieldValue.equals("")) {
+                continue;
+            }
+            if (nestedField.schema().name() != null && !nestedField.schema().name().isEmpty() && TypeConverters.containsKey(nestedField.schema().name()) && (nestedFieldValue != null && !nestedFieldValue.equals(""))) {
+                updatedNestedFieldValue = convertTimestamp(nestedFieldValue, timestampTypeFromSchema(nestedField.schema()), nestedField);
+            } else {
+                updatedNestedFieldValue = nestedFieldValue;
+            }
+            updatedNestedStructValue.put(nestedField.name(), updatedNestedFieldValue);
+        }
+        return updatedNestedStructValue;
+    }
+
+    private R applyWithSchema(R record) {
+        final Schema schema = operatingSchema(record);
+        if (config.field.isEmpty() && config.field_type.length == 0) {
+            Object value = operatingValue(record);
+            // New schema is determined by the requested target timestamp type
+            Schema updatedSchema = TRANSLATORS.get(config.type).typeSchema();
+            return newRecord(record, updatedSchema, convertTimestamp(value, timestampTypeFromSchema(schema), (Field) value));
+        } else {
+            final Struct value = requireStruct(operatingValue(record), PURPOSE);
+            Schema updatedSchema = schemaUpdateCache.get(value.schema());
```

Comment:
You're trying to get values out of the cache, but you're never putting values into the cache.

Reply:
Yeah, i am going to remove the schemaUpdateCache, as we cannot use it due to the fact that, the schema's doesn't have to be same for two records from the same table , as the schema would vary based on the null values, which we change to optional field.

So one record would have a optional field and another record from the same table would not have optional field, based on the values.